Description: Use SDL_JoystickNameForIndex to fix crash with 2 joysticks
Author: Tim Small <tim@seoss.co.uk>
Bug-Debian: https://bugs.debian.org/912171
Reviewed-by: James Cowgill <jcowgill@debian.org>
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/plugins/dfinput/cfg-gtk.c
+++ b/plugins/dfinput/cfg-gtk.c
@@ -607,7 +607,7 @@ static void PopulateDevList() {
 
 		n = SDL_NumJoysticks();
 		for (j = 0; j < n; j++) {
-			sprintf(buf, "%d: %s", j + 1, SDL_JoystickName(j));
+			sprintf(buf, "%d: %s", j + 1, SDL_JoystickNameForIndex(j));
 			gtk_list_store_append(store, &iter);
 			gtk_list_store_set(store, &iter, 0, buf, -1);
 		}
